applications/luci-app-olsr-viz/htdocs/luci-static/resources/view/olsr-viz/olsr-viz-view.js

```diff
@@ -0,0 +1,173 @@
+'use strict';
+'require uci';
+'require view';
+'require poll';
+'require ui';
+'require rpc';
+
+
+return view.extend({
+	callGetOlsrVizData: rpc.declare({
+		object: 'olsrvizinfo',
+		method: 'getolsrvizdata'
+	}),
+
+	fetch_jsoninfo: function () {
+		var jsonreq4 = '';
+		var json;
+		var data;
+		var self = this;
+		return new Promise(function (resolve, reject) {
+			L.resolveDefault(self.callGetOlsrVizData(), {})
+				.then(function (res) {
+					json = res;
+					data = json.jsonreq4;
+					resolve([data]);
+				})
+				.catch(function (err) {
+					console.error(err);
+					reject([null]);
+				});
+		});
+	},
+
+	action_olsr_viz: function () {
+		var self = this;
+		return new Promise(function (resolve, reject) {
+			self
+				.fetch_jsoninfo()
+				.then(function ([data]) {
+					var result = { viz_data: data };
+					resolve(result);
+				})
+				.catch(function (err) {
+					reject(err);
+				});
+		});
+	},
+
+	load: function () {
+		var self = this;
+		poll.add(function () {
+			self.render();
+		}, 15);
+		return new Promise(function (resolve, reject) {
+			var script = E('script', { 'type': 'text/javascript' });
+			script.onload = function () {
+				resolve();
+			};
+			script.onerror = function (err) {
+				reject(err);
+			};
+			script.src = L.resource('olsr-viz.js');
+			document.querySelector('head').appendChild(script);
+		});
+	},
+	render: function () {
+		var viz_res;
+		var self = this;
+		return this.action_olsr_viz()
+			.then(function (result) {
+				viz_res = result.viz_data;
+
+				document.querySelector('head').appendChild(E('style', { 'type': 'text/css' }, [
+					'.label {color:black;background-color:white}'
+				]));
+
+				var mainDiv = E('div', {
+					'id': 'main',
+					'style': 'width: 100%; height: 93%; border: 1px solid #ccc; margin-left:auto; margin-right:auto; text-align:center; overflow: scroll'
+				}, [
+					E('div', { 'id': 'edges', 'style': 'width: 1px; height: 1px; position: relative; z-index:2' }),
+					E('div', { 'id': 'nodes', 'style': 'width: 1px; height: 1px; position: relative; z-index:4' })
+				]);
+
+				var zoomInput = E('input', {
+					'id': 'zoom',
+					'name': 'zoom',
+					'type': 'text',
+					'value': '2.0',
+					'size': '5',
+					'style': 'min-width: unset !important;',
+					'onchange': 'set_scale()'
+				});
+				var metricInput = E('input', {
+					'id': 'maxmetric',
+					'name': 'maxmetric',
+					'type': 'text',
+					'value': '3',
+					'size': '4',
+					'style': 'min-width: unset !important;',
+					'onchange': 'set_maxmetric(this.value)'
```

Comment:
I believe this can now be `'change': (ev) => set_maxmetric(ev.target.value)`  (note the missing `on` prefix)